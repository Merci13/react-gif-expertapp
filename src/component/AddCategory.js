

import React, { useState } from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ( {setCategories} ) => {

let [inputValue, setImputValue] = useState('');


   const handleInputChange = ( e ) =>{
        setImputValue(e.target.value);
  
   } ;

   const handleSummit = ( e ) =>{
        e.preventDefault();
        if( inputValue.trim().length > 2){
              setCategories(categories => [inputValue, ...categories]);

        console.log('Summit Hecho!');
        setImputValue ("");
        }

      
   };

    return (
        <>
 
            <form onSubmit = {handleSummit}>
                  <input
            type= 'text'
            value = {inputValue}
            onChange ={ handleInputChange }
         />
            </form>
       
        </>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired

};
