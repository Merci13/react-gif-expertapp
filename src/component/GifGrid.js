
import React from 'react'
import { useFecthGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category = "" }) => {
   // const apyKey = 'wGlqhmQl8UGQDP7A0JnHkadJ1MYFb10X';





  
   
   const {data:images, loading} = useFecthGifs( category);
  



    return ( 
        <> 
        <h3 className="animate__fadeIn">{category}</h3>

        {loading && <p className= "animate__pulse"> Cargando</p>}

        <div className ="card-grid">
           

                {
                images.map((image) => (
                        // <li key = {image.id}> {image.title}</li>
                        <GifGridItem 
                        key ={image.id}
                        {...image}
                        />
                ))
                }

        </div>
        </>
   
    )
}
