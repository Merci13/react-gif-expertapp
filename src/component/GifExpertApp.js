
import React from 'react';


//import PropTypes from 'prop-types';
import { useState } from 'react/cjs/react.development';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = () =>{

    //const categories = ['One punch', 'Samura X', 'Dragon Ball'];

    const [categories, setCategories] = useState(['One punch']);


    // const handleAdd = () =>{

    //    // setCategories( [...categories, 'HunterxHunter']);
    //   //  setCategories(categories => [...categories, 'HunterxHunter']);
    
    // };

    return (
        <>
        <h2>GifExpertApp</h2>
        <hr/>
        <AddCategory setCategories = { setCategories }/>

        <hr/>

       

        <ol>
            { 
                categories.map( (category, index) => 
         ( 
         <GifGrid 
         key = {category + index}
         category = {category}/>))
            }

        </ol>

        </>
    );
};



export default GifExpertApp;