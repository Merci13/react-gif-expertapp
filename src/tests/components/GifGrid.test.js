
import { shallow } from 'enzyme';
import React from 'react';

import { GifGrid } from "../../component/GifGrid";
import { useFecthGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test Suit para GifGrid', ()=>{
    const category = 'OnePunch';
   
   

    test('Debe de mostrar el component', () => {

        useFecthGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(< GifGrid  category={ category }/>);
        expect(wrapper).toMatchSnapshot();
    });
    



    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
     
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier Cosa'
    
        },{
            id: 'ABCD',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier Cosa'
    
        }
    
    ];

         useFecthGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(< GifGrid  category={ category }/>);

          //  expect( wrapper ).toMatchSnapshot();
            expect( wrapper.find('p').exists() ).toBe(false);
            expect( wrapper.find('GifGridItem').length ).toBe( gifs.length);

    });
    



});
