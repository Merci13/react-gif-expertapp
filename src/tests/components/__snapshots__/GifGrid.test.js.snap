// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suit para GifGrid Debe de mostrar el component 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    OnePunch
  </h3>
  <p
    className="animate__pulse"
  >
     Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
